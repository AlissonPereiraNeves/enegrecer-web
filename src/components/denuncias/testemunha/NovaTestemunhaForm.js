import React from 'react';
import Telefone from '../../comum/telefone';
import * as ConstantesCSS from '../ConstantesCss';
import Genero from '../../comum/genero';
import Nome from '../../comum/nome';
import ComboboxRaca from '../../comum/comboboxRaca';
import Caracteristica from '../../comum/caracteristica';
import Data from '../../comum/data';

const NovaTestemunhaForm = () => (
  <div>
    <h3>Informacões da Testemunha</h3>
    <br />

    <div className="row">
      <Nome id={'nome-testemunha'} />
    </div>

    <div className="row">
      <Genero
        state={'generoTestemunha'}
        id={'genero-testemunha'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s6`}
      />

      <ComboboxRaca
        state={'racaTestemunha'}
        id={'raca-testemunha'}
        classes="col s6"
        somenteRacasVitima={false}
      />
    </div>

    <div className="row">
      <Data
        id={'dataNascimento-testemunha'}
        label={'Data de Nascimento'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s6`}
        state={'dataNascimentoTestemunha'}
      />

      <Telefone
        id={'telefone-vitima'}
        state={'telefoneVitima'}
        label={'Telefone'}
      />
    </div>

    <div className="row">
      <Caracteristica
        state={'caracteristicasTestemunha'}
        id={'caracteristicas-testemunha'}
        label={'* Por favor, descreva aqui as características da testemunha (máximo de 255 caracteres)'}
        type={'text'}
        inputClasse={ConstantesCSS.CLASSES_TEXTAREA}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12`}
      />
    </div>
  </div>
)

export default NovaTestemunhaForm
