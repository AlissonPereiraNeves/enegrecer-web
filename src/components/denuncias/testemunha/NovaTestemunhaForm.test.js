import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NovaTestemunhaForm from './NovaTestemunhaForm';

describe('validacoesCamposTestemunha', () => {
  it('renderiza o componente sem erros', () => {
    const wrapper = shallow(<NovaTestemunhaForm />);
    const wrapperJson = toJson(wrapper);
    expect(wrapperJson).toMatchSnapshot();
  });
});
