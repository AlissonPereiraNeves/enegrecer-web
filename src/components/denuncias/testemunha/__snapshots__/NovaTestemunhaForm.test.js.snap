// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validacoesCamposTestemunha renderiza o componente sem erros 1`] = `
<div>
  <h3>
    Informacões da Testemunha
  </h3>
  <br />
  <div
    className="row"
  >
    <nome
      id="nome-testemunha"
      onChange={[Function]}
      state=""
    />
  </div>
  <div
    className="row"
  >
    <genero
      divClasse="input-field col s6"
      id="genero-testemunha"
      state="generoTestemunha"
    />
    <comboRaca
      classes="col s6"
      id="raca-testemunha"
      somenteRacasVitima={false}
      state="racaTestemunha"
    />
  </div>
  <div
    className="row"
  >
    <Data
      divClasse="input-field col s6"
      id="dataNascimento-testemunha"
      label="Data de Nascimento"
      state="dataNascimentoTestemunha"
    />
    <telefone
      divClasse=""
      id="telefone-vitima"
      label="Telefone"
      state="telefoneVitima"
    />
  </div>
  <div
    className="row"
  >
    <caracteristica
      divClasse="input-field col s12"
      id="caracteristicas-testemunha"
      inputClasse="materialize-textarea"
      label="* Por favor, descreva aqui as características da testemunha (máximo de 255 caracteres)"
      state="caracteristicasTestemunha"
      type="text"
    />
  </div>
</div>
`;
